### PR TITLE
Add jdbc analyzer for maximum

### PR DESCRIPTION
**Pull request by janwestphal306**
*Issue #1*

*Description of changes:*
Added the jdbc analyzer for maximum.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import com.amazon.deequ.metrics.{DoubleMetric, Entity}
+import org.postgresql.util.PSQLException
+
+case class JdbcMaximum(column: String)
+  extends JdbcAnalyzer[MaxState, DoubleMetric] {
+
+  override def validateParams(table: Table, column: String): Unit = {
+    super.validateParams(table, column)
+
+    isNumeric(table, column)
```
Comment (mschroederi) [inline src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala]: Include `import com.amazon.deequ.analyzers.jdbc.Preconditions.isNumeric`

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    catch {
+      case error: Exception => throw error
+    }
+  }
+
+  override def computeMetricFrom(state: Option[MaxState]): DoubleMetric = {
+    state match {
+      case Some(theState) =>
+        metricFromValue(theState.metricValue(), "MaxState", column, Entity.Column)
```
Comment (mschroederi) [inline src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala]: I think "Maximum" is the right option here (not "MaxState")

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      case _ =>
+        toFailureMetric(new EmptyStateException(
+          s"Empty state for analyzer JdbcMaximum, all input values were NULL."))
+    }
+  }
+
+  override private[deequ] def toFailureMetric(failure: Exception) = {
+    metricFromFailure(failure, "MaxState", column, Entity.Column)
```
Comment (mschroederi) [inline src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala]: see above

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala (hunk 1 above)
Comment (janwestphal306) [reply src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala]: Fixed

**Pull request merged**